CardParts/src/Classes/Card Parts/CardPartIconLabel.swift

```diff
@@ -0,0 +1,166 @@
+                iconView.frame = iconView.frame.offsetBy(dx: padding, dy: iconYPosition)
+                newRect = CGRect(x: iconView.frame.width + iconPadding + padding, y: 0, width: frame.width - (iconView.frame.width + iconPadding), height: height)
+            }else if textAlignment == .right {
+                iconView.frame = iconView.frame.offsetBy(dx: frame.width - size.width - iconPadding - iconView.frame.width - padding, dy: iconYPosition)
+                newRect = CGRect(x: frame.width - size.width - iconPadding - padding , y: 0, width: size.width + iconPadding, height: height)
+            }else if textAlignment == .center {
+                iconView.frame = iconView.frame.offsetBy(dx: (frame.width - size.width) / 2 - iconPadding - iconView.frame.width, dy: iconYPosition)
+                newRect = CGRect(x: (frame.width - size.width) / 2 , y: 0, width: size.width + iconPadding, height: height)
+            }
+        case .right:
+            if textAlignment == .left {
+                iconView.frame = iconView.frame.offsetBy(dx: size.width + iconPadding + padding, dy: iconYPosition)
+                newRect = CGRect(x: 10, y: 0, width: size.width, height: height)
+            }else if textAlignment == .right {
+                iconView.frame = iconView.frame.offsetBy(dx: frame.width - iconView.frame.width - padding , dy: iconYPosition)
+                newRect = CGRect(x: frame.width - size.width - iconView.frame.width - iconPadding - 2 * padding, y: 0, width: size.width + iconPadding, height: height)
+            }else if textAlignment == .center {
+                iconView.frame = iconView.frame.offsetBy(dx: frame.width / 2 + size.width / 2 + iconPadding, dy: iconYPosition)
+                newRect = CGRect(x: (frame.width - size.width) / 2, y: 0, width: size.width, height: height)
+            }
+        }
+        
+        // set the frame positions of the icon vertically i.e., top, bottom or center
+        switch iconPosition.vertical {
+        case .top: iconView.frame.origin.y = (frame.height - size.height) / 2
+            
+        case .center: iconView.frame.origin.y = (frame.height - iconView.frame.height) / 2
+            
+        case .bottom: iconView.frame.origin.y = frame.height - (frame.height - size.height) / 2 - iconView.frame.size.height
+        }
+        
+        addSubview(iconView)
+        super.drawText(in: newRect)
+        setup()
+    }
+    
+    /// returns new size with updated vertical and horizantal padding.
+    public override var intrinsicContentSize: CGSize {
+        let superSize = super.intrinsicContentSize
+        let newWidth = superSize.width + superSize.height + (2 * horizontalPadding)
+        let newHeight = superSize.height + (2 * verticalPadding)
+        let size = CGSize(width: newWidth, height: newHeight)
+        return size
+    }
+}
+
+// MARK: - Reactive binding for label's text , vertical & horizontal padding.
+extension Reactive where Base: CardPartIconLabel {
```

Comment:
Do we want to add bindability for the image?

Reply:
done.